test/plugin_functional/test_suites/core_plugins/rendering.ts

```diff
@@ -284,6 +284,11 @@ export default function ({ getService }: PluginFunctionalProviderContext) {
         'xpack.observability.unsafe.alertDetails.uptime.enabled (boolean)',
         'xpack.observability.unsafe.thresholdRule.enabled (boolean)',
         'xpack.observability_onboarding.ui.enabled (boolean)',
+        /**
+         * xpack.discoverLogExplo.featureFlags is conditional and will resolve to an object of properties
+         * - xpack.discoverLogExplo.featureFlags.deepLinkVisible (boolean)
+         */
+        'xpack.discoverLogExplorer.featureFlags (any)',
```

Comment:
__note:__ I'm surprised that it's `xpack.discoverLogExplorer.featureFlags` and not `xpack.discoverLogExplorer.featureFlags.deepLinkVisible` even though only specific field is exposed:
```ts
  exposeToBrowser: {
    featureFlags: {
      deepLinkVisible: true,
    },
  },
```
@elastic/kibana-core is it expected behavior for the case when there is only one sub-field? If so, will adding more fields to `xpack.discoverLogExplorer.featureFlags` trigger change in this file (so that we can review them)?

Reply:
That's a side-effect of using `never` at type for `deepLinkVisible ` when not on serverless:

```ts
export const configSchema = schema.object({
  featureFlags: schema.object({
    deepLinkVisible: schema.conditional(
      schema.contextRef('serverless'),
      true,
      schema.boolean(),
      schema.never(),
      {
        defaultValue: false,
      }
    ),
  }),
});
```

 So when `serverless` is `false`, the config schema basically is:

```ts
schema.object({
  featureFlags: schema.object({})
})
```

So `featureFlags` is considered a leaf, given it doesn't have any nested properties, causing our enumeration logic to list `xpack.discoverLogExplorer.featureFlags` as a leaf.

Reply:
So probably not an expected behavior, although last time I asked @lukeelmers, using this `never` approach was still the recommended way of doing this... So... Yeah.

Reply:
Got it, thanks for clarifying that @pgayvallet! I see the the approach [has changed slightly](https://github.com/elastic/kibana/pull/161939/commits/fc1eb0cc16b4a91abe99b9d5fd6a1ba8269794a0) since my comment, and now we have `xpack.discoverLogExplorer.featureFlags.deepLinkVisible` instead of `xpack.discoverLogExplorer.featureFlags` in the test, which works for me. Thanks for also adding a comment @tonyghiani!